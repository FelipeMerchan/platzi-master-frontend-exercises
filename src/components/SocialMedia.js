import React from 'react'

import './styles/SocialMedia.css'
import facebook from '../images/social-media/facebook.png'
import digg from '../images/social-media/digg.png'
import flickr from '../images/social-media/flickr.png'
import twitter from '../images/social-media/twitter.png'

function SocialMedia() {
  return (
    <article className="SocialMedia">
      <h2 className="SocialMedia__title">tambi√©n
        <br /> <span>estoy en</span>
      </h2>
      <a href="https://www.facebook.com/" target="_blank" rel="noopener noreferrer">
        <img className="SocialMedia__icon" width="30" src={facebook} alt="Facebook" />
        <strong className="SocialMedia__url">www.facebook.com/lorem</strong>
      </a>
      <a href="https://www.facebook.com/" target="_blank" rel="noopener noreferrer">
        <img className="SocialMedia__icon" width="30" src={digg} alt="Facebook" />
        <strong className="SocialMedia__url">www.digg.com/user/lorem</strong>
      </a>
      <a href="https://www.facebook.com/" target="_blank" rel="noopener noreferrer">
        <img className="SocialMedia__icon" width="30" src={flickr} alt="Facebook" />
        <strong className="SocialMedia__url">www.flickr.com/photos/lorem</strong>
      </a>
      <a href="https://www.facebook.com/" target="_blank" rel="noopener noreferrer">
        <img className="SocialMedia__icon" width="30" src={twitter} alt="Facebook" />
        <strong className="SocialMedia__url">www.twitter.com/lorem</strong>
      </a>
    </article>
  )
}

export default SocialMedia