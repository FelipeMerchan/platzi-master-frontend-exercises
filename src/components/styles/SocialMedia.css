.SocialMedia__title {
  margin-block-end: var(--space-2);
  font-family: var(--font-4);
  font-size: 40px;
  color: #c5345c;
  line-height: .7;
}

.SocialMedia__title>span {
  color: #66615f;
  padding-inline-start: var(--space-5);
}

.SocialMedia a {
  display: block;
  margin-block-end: var(--space);
}

.SocialMedia__url {
  margin-inline-start: var(--space);
  color: #c5345c;
}