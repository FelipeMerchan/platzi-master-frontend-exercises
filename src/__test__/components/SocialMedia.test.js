import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import SocialMedia from '../../components/SocialMedia'

describe('Probar el componente <SocialMedia />', () => {
  const socialMedia = shallow(<SocialMedia />)
  test('Render del componente SocialMedia', () => {
    expect(socialMedia.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(socialMedia.exists('.SocialMedia__title')).toBeTruthy()
  })
  test('Render de todas las RRSS en las que se tiene presencia', () => {
    expect(socialMedia.children()).toHaveLength(5);
  });
})

describe('SocialMedia Snapshot', () => {
  const socialMedia = create(<SocialMedia />)
  test('Comprobar la UI del componente SocialMedia', () => {
    expect(socialMedia.toJSON()).toMatchSnapshot()
  })
})