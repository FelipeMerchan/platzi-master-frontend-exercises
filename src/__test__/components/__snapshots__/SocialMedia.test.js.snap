// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia Snapshot Comprobar la UI del componente SocialMedia 1`] = `
<article
  className="SocialMedia"
>
  <h2
    className="SocialMedia__title"
  >
    tambi√©n
    <br />
     
    <span>
      estoy en
    </span>
  </h2>
  <a
    href="https://www.facebook.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Facebook"
      className="SocialMedia__icon"
      src="facebook.png"
      width="30"
    />
    <strong
      className="SocialMedia__url"
    >
      www.facebook.com/lorem
    </strong>
  </a>
  <a
    href="https://www.facebook.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Facebook"
      className="SocialMedia__icon"
      src="digg.png"
      width="30"
    />
    <strong
      className="SocialMedia__url"
    >
      www.digg.com/user/lorem
    </strong>
  </a>
  <a
    href="https://www.facebook.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Facebook"
      className="SocialMedia__icon"
      src="flickr.png"
      width="30"
    />
    <strong
      className="SocialMedia__url"
    >
      www.flickr.com/photos/lorem
    </strong>
  </a>
  <a
    href="https://www.facebook.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Facebook"
      className="SocialMedia__icon"
      src="twitter.png"
      width="30"
    />
    <strong
      className="SocialMedia__url"
    >
      www.twitter.com/lorem
    </strong>
  </a>
</article>
`;
